demo-apps/cf-android/app/src/main/java/org/eclipse/californium/examples/MainActivity.java

```diff
@@ -103,4 +123,19 @@ protected void onPostExecute(CoapResponse response) {
             }
         }
     }
+
+    private void initCoapEndpoint(String uri){
+        CoapEndpoint.CoapEndpointBuilder endpointBuilder = new CoapEndpoint.CoapEndpointBuilder();
+        if(uri.startsWith(CoAP.COAP_SECURE_URI_SCHEME+"://")){
+            DtlsConnectorConfig.Builder dtlsConfig = new DtlsConnectorConfig.Builder();
+            dtlsConfig.setClientOnly();
+            ConfigureDtls.loadCredentials(getApplicationContext(), dtlsConfig, CLIENT_NAME);
+            DTLSConnector dtlsConnector = new DTLSConnector(dtlsConfig.build());
+            endpointBuilder.setConnector(dtlsConnector);
+        }else if(uri.startsWith(CoAP.COAP_URI_SCHEME+"://")){
+            UDPConnector connector = new UDPConnector();
+            endpointBuilder.setConnector(connector);
+        }
+        EndpointManager.getEndpointManager().setDefaultEndpoint(endpointBuilder.build());
+    }
```

Comment:
Please, read my comment https://github.com/eclipse/californium/pull/634#issuecomment-388541186! 

Your idea may seem to work, but it's too complicated and not as intended. The most examples of californium usually just create a endpoint use it and drop it on termination. So these other examples are not too useful in this case. Therefore I recommend to setup the EndpointManager with both plain and secure endpoints, and then just provide the url to the coap client, and let CoapClient.getEffectiveEndpoint(Request) do the job. When you change your code to that pattern, you will see, how easy it is :-).